jobs/component_chart_publish.groovy

```diff
@@ -59,45 +59,46 @@ repos.each { Map repo ->
       }
 
       steps {
-        shell """
-          #!/usr/bin/env bash
-
-          set -eo pipefail
-          set -x
-
-          # check out RELEASE_TAG tag
-          commit="\$(git checkout "\${RELEASE_TAG}" && git rev-parse HEAD)"
-
-          if [ -d charts ]; then
-            cd charts
-            ${defaults.helm.downloadAndInit}
-
-            ## change chart values
-            # update the chart version to RELEASE_TAG
-            perl -i -0pe "s/<Will be populated by the ci before publishing the chart>/\${RELEASE_TAG}/g" ${repo.chart}/Chart.yaml
-            # update all org values to "deis"
-            perl -i -0pe 's/"deisci"/"deis"/g' ${repo.chart}/values.yaml
-            # update the image pull policy to "IfNotPresent"
-            perl -i -0pe 's/"Always"/"IfNotPresent"/g' ${repo.chart}/values.yaml
-            # update the dockerTag value to RELEASE_TAG
-            perl -i -0pe "s/canary/\${RELEASE_TAG}/g" ${repo.chart}/values.yaml
-
-            # package release chart
-            helm package ${repo.chart}
-
-            # download index file from aws s3 bucket
-            aws s3 cp s3://helm-charts/${repo.chart}/index.yaml .
-
-            # update index file
-            helm repo index . --url https://charts.deis.com/${repo.chart} --merge ./index.yaml
-
-            # push packaged chart and updated index file to aws s3 bucket
-            aws s3 cp ${repo.chart}-\${RELEASE_TAG}.tgz s3://helm-charts/${repo.chart}/ \
-              && aws s3 cp index.yaml s3://helm-charts/${repo.chart}/index.yaml
-          else
-            echo "No 'charts' directory found at project level; nothing to publish."
-          fi
-        """.stripIndent().trim()
+        shell new File("${workspace}/bash/scripts/helm_chart_actions.sh").text +
+          """
+            #!/usr/bin/env bash
+
+            set -eo pipefail
+            set -x
+
+            # check out RELEASE_TAG tag
+            commit="\$(git checkout "\${RELEASE_TAG}" && git rev-parse HEAD)"
+
+            if [ -d charts ]; then
+              cd charts
```

Comment:
It doesn't look like this script does a `cd ..` later, so probably `pushd/popd` aren't necessary here.
https://en.wikipedia.org/wiki/Pushd_and_popd